frontend/src/components/pod/Details.tsx

```diff
@@ -96,38 +97,36 @@ function PodLogViewer(props: PodLogViewerProps) {
   }
   const debouncedSetState = _.debounce(setLogsDebounced, 500, options);
 
-  React.useEffect(
-    () => {
-      let callback: any = null;
+  React.useEffect(() => {
+    let callback: any = null;
 
-      if (props.open) {
-        xtermRef.current?.clear();
-        setLogs({ logs: [], lastLineShown: -1 });
+    if (props.open) {
+      xtermRef.current?.clear();
+      setLogs({ logs: [], lastLineShown: -1 });
 
-        callback = item.getLogs(container, debouncedSetState, {
-          tailLines: lines,
-          showPrevious,
-          showTimestamps,
-          follow,
-        });
-      }
+      callback = item.getLogs(container, debouncedSetState, {
+        tailLines: showAll ? 'all' : parseInt(lines.toString(), 10) || 100,
+        showPrevious,
+        showTimestamps,
+        follow,
+      });
+    }
 
-      return function cleanup() {
-        if (callback) {
-          callback();
-        }
-      };
-    },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [container, lines, open, showPrevious, showTimestamps, follow]
-  );
+    return function cleanup() {
+      if (callback) {
+        callback();
+      }
+    };
+  }, [container, lines, props.open, showPrevious, showTimestamps, follow, showAll]);
 
   function handleContainerChange(event: any) {
     setContainer(event.target.value);
   }
 
   function handleLinesChange(event: any) {
-    setLines(event.target.value);
+    const value = event.target.value;
+    setShowAll(value === 'all');
+    setLines(showAll ? 0 : parseInt(value, 10) || 100);
```

Comment:
If we use negative numbers as the way to set up the "all", then we don't need an extra state var and the logic becomes easier IMO.